src/logicblocks/event/processing/broker/strategies/singleton/factories.py

```diff
@@ -20,41 +17,45 @@
 
 
 class InMemorySingletonEventBrokerBuilder(
-    SingletonEventBrokerBuilder[(InMemoryEventStorageAdapter,)]
+    SingletonEventBrokerBuilder[(EventStorageAdapter,)]
 ):
     def dependencies(
-        self, adapter: InMemoryEventStorageAdapter
+        self, adapter: EventStorageAdapter
     ) -> SingletonEventBrokerDependencies:
         return SingletonEventBrokerDependencies(
-            event_source_factory=InMemoryEventStoreEventSourceFactory(
-                adapter=adapter
-            )
+            event_source_factory=EventStoreEventSourceFactory(adapter=adapter)
         )
 
 
 class PostgresSingletonEventBrokerBuilder(
     SingletonEventBrokerBuilder[
-        (ConnectionSettings, AsyncConnectionPool[AsyncConnection])
+        (
+            ConnectionSettings,
+            AsyncConnectionPool[AsyncConnection],
+            EventStorageAdapter | None,
+        )
     ]
 ):
     def dependencies(
         self,
         connection_settings: ConnectionSettings,
         connection_pool: AsyncConnectionPool[AsyncConnection],
+        adapter: EventStorageAdapter | None = None,
     ) -> SingletonEventBrokerDependencies:
+        event_storage_adapter = adapter or PostgresEventStorageAdapter(
```

Comment:
This should strictly be `adapter if adapter is not None else Postgres...`. In this instance it doesn't really matter based on the only valid value (according to the type) of adapter is an `EventStorageAdapter` but in general `or` will treat falsey but valid values on the left as not valid. So in general, I have been being explicit about None.